Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +222,253 @@ function onInterval(): void {
+}
+
+function isArrayOfAssets(input: any): input is Asset[] {
+    if (!(input instanceof Array)) {
+        return false;
+    }
+    for (const item of input) {
+        if (!isAsset(item)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+function isReleaseJson(input: any): input is Build[] {
+    if (!input || !(input instanceof Array) || input.length === 0) {
+        return false;
+    }
+    return isBuild(input[0]);
+}
+
+async function installVsix(vsixLocation: string, updateChannel: string): Promise<void> {
+    // Get the path to the VSCode command -- replace logic later when VSCode allows calling of
+    // workbench.extensions.action.installVSIX from TypeScript w/o instead popping up a file dialog
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsCodeScriptPath: string | undefined = await async function(platformInfo): Promise<string | undefined> {
+        if (platformInfo.platform === 'win32') {
+            const vsCodeProcessPath: string = path.dirname(process.execPath);
+            return path.join(vsCodeProcessPath, 'bin', 'code.cmd');
+        } else {
+            const stdout: Buffer = await execSync("which code");
+            return stdout.toString().trim();
+        }
+    }(platformInfo);
+    if (!vsCodeScriptPath) {
+        telemetry.logLanguageServerEvent('findVsCodeScriptFailure');
+        return;
+    }
+
+    // Install the VSIX
+    const installCommand: string = vsCodeScriptPath + ' --install-extension ' + vsixLocation;
+    try {
+        if (updateChannel === 'Default') {
+            // Uninstall the current version, as the version to install is a previous version
+            const unInstallCommand: string = vsCodeScriptPath + ' --uninstall-extension ms-vscode.cpptools';
+            await execSync(unInstallCommand);
+        }
+        await execSync(installCommand);
+    } catch (error) {
+        telemetry.logLanguageServerEvent("installCommandFailure");
+    }
+}
+
+async function downloadUrlForPlatform(build: Build): Promise<string | undefined> {
+    // Get the VSIX name to search for in build
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsixName: string | undefined = function(platformInfo): string {
+        switch (platformInfo.platform) {
+            case 'win32': return 'cpptools-win32.vsix';
+            case 'darwin': return 'cpptools-osx.vsix';
+            default: {
+                switch (platformInfo.architecture) {
+                    case 'x86': return 'cpptools-linux32.vsix';
+                    case 'x86_64': return 'cpptools-linux.vsix';
+                }
+            }
+        }
+    }(platformInfo);
+    if (!vsixName) {
+        return;
+    }
+
+    // Get the URL to download the VSIX, using vsixName as a key
+    const downloadUrl: string | undefined = build.assets.find((asset) => {
```

Comment:
when you have a single-line lambda that returns a result, you should use this syntax:
```
build.assets.find(asset => asset.name === vsixName);
```